pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -679,18 +719,51 @@ async def get_instructions(run_context: RunContext[AgentDepsT]) -> str | None:
                 if instrumentation_settings and run_span.is_recording():
                     run_span.set_attributes(
                         self._run_span_end_attributes(
-                            instrumentation_settings, usage, state.message_history, graph_deps.new_message_index
+                            instrumentation_settings,
+                            usage,
+                            state.message_history,
+                            graph_deps.new_message_index,
+                            run_metadata,
                         )
                     )
             finally:
                 run_span.end()
 
+    def _compute_agent_metadata(
```

Comment:
For consistency with the other fields supported by `agent.override`, let's call this `_get_metadata()` and the `run_metadata_config` `additional_metadata`